.buildkite/pipeline.yml

```diff
@@ -11,125 +11,149 @@ env:
+  #    plugins:
+  #      - test-collector#v1.2.0:
+  #          files: "lib/*/unittest/boost_test_results.junit"
+  #          format: "junit"
+  #
+  #  - label: "Build & test :cpp: for linux-aarch64-RelWithDebInfo :linux: :aws:"
+  #    timeout_in_minutes: 180
+  #    agents:
+  #      provider: aws
+  #      instanceType: m6g.2xlarge
+  #      imagePrefix: ci-amazonlinux-2-aarch64
+  #      diskSizeGb: 100
+  #      diskName: '/dev/xvda'
+  #    commands:
+  #      - ".buildkite/scripts/steps/build_and_test.sh"
+  #    depends_on: "check_style"
+  #    key: "build_test_linux-aarch64-RelWithDebInfo"
+  #    artifact_paths:
+  #      - "lib/*/unittest/boost_test_results.junit"
+  #      - "bin/*/unittest/boost_test_results.junit"
+  #    env:
+  #      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=/ml-cpp/cmake/linux-aarch64.cmake"
+  #      CPP_CROSS_COMPILE: ""
+  #      RUN_TESTS: "true"
+  #      BOOST_TEST_OUTPUT_FORMAT_FLAGS: "--logger=JUNIT,error,boost_test_results.junit"
+  #    plugins:
+  #      - test-collector#v1.2.0:
+  #          files: "lib/*/unittest/boost_test_results.junit"
+  #          format: "junit"
+  #
+  #  - label: "Build :cpp: for linux_aarch64_cross-RelWithDebInfo :linux:"
+  #    timeout_in_minutes: 180
+  #    agents:
+  #      cpu: "6"
+  #      ephemeralStorage: "20G"
+  #      memory: "64G"
+  #      image: "docker.elastic.co/ml-dev/ml-linux-aarch64-cross-build:10"
+  #    commands:
+  #      - ".buildkite/scripts/steps/build_and_test.sh"
+  #    depends_on: "check_style"
+  #    key: "build_linux_aarch64_cross-RelWithDebInfo"
+  #    env:
+  #      CPP_CROSS_COMPILE: "aarch64"
+  #      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=cmake/linux-aarch64.cmake"
+  #      RUN_TESTS: "false"
+  #
+  #  - label: "Build :cpp: for macos_aarch64_cross-RelWithDebInfo :macos:"
+  #    timeout_in_minutes: 180
+  #    agents:
+  #      cpu: "6"
+  #      ephemeralStorage: "20G"
+  #      memory: "64G"
+  #      image: "docker.elastic.co/ml-dev/ml-macosx-build:16"
+  #    commands:
+  #      - ".buildkite/scripts/steps/build_and_test.sh"
+  #    depends_on: "check_style"
+  #    key: "build_macos_aarch64_cross-RelWithDebInfo"
+  #    env:
+  #      CPP_CROSS_COMPILE: "macosx"
+  #      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=cmake/darwin-x86_64.cmake"
+  #      RUN_TESTS: "false"
+  #
+  #
+  - label: "Build :cpp: for Windows_x86_64-RelWithDebInfo :windows:"
     timeout_in_minutes: 180
     agents:
-      provider: aws
-      instanceType: m6g.2xlarge
-      imagePrefix: ci-amazonlinux-2-aarch64
-      diskSizeGb: 100
-      diskName: '/dev/xvda'
+      provider: gcp
+      image: family/ml-cpp-1-windows-2016
     commands:
-      - ".buildkite/scripts/steps/build_and_test.sh"
-    depends_on: "check_style"
-    key: "build_test_linux-aarch64-RelWithDebInfo"
-    artifact_paths:
-      - "lib/*/unittest/boost_test_results.junit"
-      - "bin/*/unittest/boost_test_results.junit"
```

Comment:
On Linux we're preserving the JUnit output as artifacts as well as passing to the test collector. I think we should do this on Windows too. Anything we can keep to make understanding test failures easier is good. If it's not possible to have the same files be artifacts and also be passed to the test collector then maybe we could make the Boost XML output format the artifacts and pass the JUnit format to the test collector.

Also, it seems like it should be relatively simple to make all the unit test log files artifacts. Currently we `cat` the log for any test suite that fails, but it's not possible to view it without the risk of output from parallel processes being mixed in, and it's not possible to view the output of other test suites that ran simultaneously to look for possible interactions. So it would be nice if all the unit test logs got preserved (for all platforms, not just Windows).